File: src/ddf/rule.py
"""A Rule for a field has Actions."""

from .action import Action
from .js import JsDictMixin


class Rule(JsDictMixin):
    """A rule applies actions if constraints pass."""

    js_attrs = ['actions', 'field']

    def __init__(self, field, actions):
        """List the actions to apply on the field."""
        self.actions = actions
        self.field = field

        for action in actions:
            if isinstance(action, Action):
                continue
            raise Exception('%s is not an Action' % action)

    def apply(self, form):
        """Execute each action on the form."""
        applied = []
        for action in self.actions:
            if action.execute(form.fields[self.field]):
                applied.append(action)
        return applied

File: src/ddf/rule.js
import debug from 'debug'

var log = debug('ddf.rule')

// A Rule has actions for a field.
class Rule {
  constructor(field, actions) {
    this.field = field
    this.actions = actions
  }

  // Execute all actions of this rule.
  apply(form) {
    log('[Rule]', this, '.apply(', form, ')')
    for (var i=0; i<this.actions.length; i++) {
      this.actions[i].execute(form.field(this.field))
    }
  }
}

export {
  Rule,
}
